src/Core/Services/IProviderService.cs

```diff
@@ -28,5 +28,6 @@ public interface IProviderService
         Task<ProviderOrganization> CreateOrganizationAsync(Guid providerId, OrganizationSignup organizationSignup,
             string clientOwnerEmail, User user);
         Task RemoveOrganization(Guid providerId, Guid providerOrganizationId, Guid removingUserId);
+        Task LogProviderAccessToOrganization(Guid organizationId);
```

Comment:
As the other async methods name append an "Async" , I'd suggest adding them here and although not part of it also to RemoveOrganization